be/src/exec/pipeline/aggregate_blocking_operator.cpp

```diff
@@ -0,0 +1,124 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "aggregate_blocking_operator.h"
+
+#include "column/chunk.h"
+#include "column/column.h"
+#include "column/column_helper.h"
+#include "column/const_column.h"
+#include "config.h"
+#include "exprs/expr_context.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks::pipeline {
+
+void AggregateBlockingOperator::finish(RuntimeState* state) {
+    if (_is_pre_finished) return;
```

Comment:
finish is invoked exactly once

Reply:
finish may invoke many times in the entire life cycle